### PR TITLE
Create Julia 1.2 version of test project

### PR DESCRIPTION
**Pull request by daniel-thom**
This adds logging during tests as well as the ability to run individual tests.  It doesn't yet load the full test environment unless you call Pkg.test().  I'll add that as a separate PR.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -13,6 +14,8 @@ using Cbc
 using OSQP
 using TimeSeries
 using ParameterJuMP
+using Compat: Test
```
Comment (jd-lara) [inline test/runtests.jl]: Why do we need to use compat here to import the Test package?

Comment (daniel-thom) [reply test/runtests.jl]: Per the TestSetExtensions docs:  https://github.com/ssfrr/TestSetExtensions.jl

Comment (jd-lara) [reply test/runtests.jl]: I don't think that's necessary, the README hasn't been updated since 2017

Comment (daniel-thom) [reply test/runtests.jl]: You are correct.  Thanks.

**Pull request merged**